packages/shared/src/hooks/useConditionalFeature.ts

```diff
@@ -0,0 +1,36 @@
+import { useQuery } from '@tanstack/react-query';
+import { JSONValue, WidenPrimitives } from '@growthbook/growthbook';
+import { useContext } from 'react';
+import { generateQueryKey, RequestKey } from '../lib/query';
+import { Feature } from '../lib/featureManagement';
+import { disabledRefetch } from '../lib/func';
+import AuthContext from '../contexts/AuthContext';
+import { useGrowthBookContext } from '../components/GrowthBookProvider';
+
+export const useConditionalFeature = <T extends JSONValue>({
+  feature,
+  shouldEvaluate,
+}: {
+  feature: Feature<T>;
+  shouldEvaluate: boolean;
+}): WidenPrimitives<T> => {
+  const { user } = useContext(AuthContext);
+  const { growthbook } = useGrowthBookContext();
+  const queryKey = generateQueryKey(RequestKey.Feature, user, feature.id);
+
+  const { data: featureValue } = useQuery(
+    queryKey,
+    () => {
+      return growthbook
+        ? growthbook.getFeatureValue(feature.id, feature.defaultValue)
+        : (feature.defaultValue as WidenPrimitives<T>);
+    },
+    {
+      initialData: feature.defaultValue as WidenPrimitives<T>,
```

Comment:
Shouldn't this be optional? Some cases would want to wait for the evaluation I guess

Reply:
User is defaultValue (control) until proven different right? But loading state makes sense. Maybe we can also add an argument to hook for defaultValue so we can fallback it when needed?

Reply:
ok, again we just have to watch out from flickering and layout shifts